src/schemes/fluid/transport_velocity.jl

```diff
@@ -0,0 +1,178 @@
+        end
+    end
+
+    return v0
+end
+
+# Add momentum velocity.
+@inline function add_velocity!(du, v, particle, system, ::Nothing)
+    for i in 1:ndims(system)
+        du[i, particle] = v[i, particle]
+    end
+
+    return du
+end
+
+# Add advection velocity.
+@inline function add_velocity!(du, v, particle, system, ::TransportVelocityAdami)
+    for i in 1:ndims(system)
+        du[i, particle] = v[ndims(system) + i, particle]
+    end
+
+    return du
+end
+
+@inline function advection_velocity(v, system, particle)
+    return SVector(ntuple(@inline(dim->v[ndims(system) + dim, particle]), ndims(system)))
+end
+
+@inline function momentum_convection(system, neighbor_system, ::Nothing,
+                                     v_particle_system, v_neighbor_system, rho_a, rho_b,
+                                     m_a, m_b, particle, neighbor, grad_kernel)
+    return SVector(ntuple(_ -> 0.0, Val(ndims(system))))
+end
+
+@inline function momentum_convection(system, neighbor_system, ::TransportVelocityAdami,
+                                     v_particle_system, v_neighbor_system, rho_a, rho_b,
+                                     m_a, m_b, particle, neighbor, grad_kernel)
+    volume_a = m_a / rho_a
+    volume_b = m_b / rho_b
+    volume_term = (volume_a^2 + volume_b^2) / m_a
+
+    momentum_velocity_a = current_velocity(v_particle_system, system, particle)
+    advection_velocity_a = advection_velocity(v_particle_system, system, particle)
+
+    momentum_velocity_b = current_velocity(v_neighbor_system, neighbor_system, neighbor)
+    advection_velocity_b = advection_velocity(v_neighbor_system, neighbor_system, neighbor)
+
+    A_a = rho_a * momentum_velocity_a * (advection_velocity_a - momentum_velocity_a)'
+    A_b = rho_b * momentum_velocity_b * (advection_velocity_b - momentum_velocity_b)'
+
+    return volume_term * (0.5 * (A_a + A_b)) * grad_kernel
+end
+
+@inline transport_velocity!(dv, system, rho_a, rho_b, m_a, m_b, grad_kernel, particle) = dv
+
+@inline function transport_velocity!(dv, system::FluidSystem,
+                                     rho_a, rho_b, m_a, m_b, grad_kernel, particle)
+    transport_velocity!(dv, system, system.transport_velocity, rho_a, rho_b, m_a, m_b,
+                        grad_kernel, particle)
+end
+
+@inline function transport_velocity!(dv, system, ::Nothing,
+                                     rho_a, rho_b, m_a, m_b, grad_kernel, particle)
+    return dv
+end
+
+@inline function transport_velocity!(dv, system, ::TransportVelocityAdami,
+                                     rho_a, rho_b, m_a, m_b, grad_kernel, particle)
+    (; transport_velocity) = system
+    (; background_pressure) = transport_velocity
+
+    NDIMS = ndims(system)
+
+    volume_a = m_a / rho_a
+    volume_b = m_b / rho_b
+    volume_term = (volume_a^2 + volume_b^2) / m_a
+
+    for dim in 1:NDIMS
+        dv[NDIMS + dim, particle] -= volume_term * background_pressure * grad_kernel[dim]
+    end
+
+    return dv
+end
```

Comment:
I'm pretty sure this doesn't work as intended.
Say we're using the Euler method for time integration.
Then we have only one stage. And at the beginning of this stage, velocity and transport velocity are the same, since you just did that in the update callback.

Maybe it's time to ask Adami if there is a way to implement this for multi-stage time integration methods?

Reply:
It is an explicit scheme so any suitable integrator can be used, no?
I mean, when using the euler method, it is just a normal evaluation without TVF (without correction). It's not  incorrect then, no? 

On the other hand, when using a multi-stage integration scheme, we correct the momentum at each stage as it is supposed to be. 

Nevertheless you're right. We should contact Adami.

Reply:
The correction is applied during the stages. For momentum_velocity = advection_velocity (single stage Euler method) the convection term is zero. That is, no correction due to transport velocity.

Reply:
So far I understood, the TVF only takes affect for integration schemes with 2 stages and more.

Reply:
The kick-drift-kick form used in Adami's paper is basically only one stage. It never uses the non-transport velocity for advection, only the transport velocity. Here, the first stage would use the regular velocity.

Reply:
>  It never uses the non-transport velocity for advection, only the transport velocity.

I'm not sure what you mean, but, yes, the position update always uses the transport velocity.

> Here, the first stage would use the regular velocity.

Yes, at first stage, no correction occurs, since transport velocity is equal to momentum velocity. But during the stages, the correction is applied due to the transport velocity. 
So to speak, the particles are forced to be on the supposed trajectory during the stages according to the background pressure. 

[Ramachandran (2019)](https://www.sciencedirect.com/science/article/pii/S0045793018309046?via%3Dihub) is using a two-stage Evaluate-Predict-Evaluate-Correct (EPEC) scheme.

Reply:
This is from Ramachandran 2019:
<img width="561" alt="grafik" src="https://github.com/user-attachments/assets/2e218110-b00c-4f57-af2a-4c9235c10cf9">
Note that in the first stage, the transport velocity is used to advect the particles, while in your implementation, the regular velocity is used in the first stage.

Thank you for pointing me to this article. I've been thinking a lot about how to correctly implement this for multi-stage methods. So the correct solution is to just compute the transport velocity from the regular velocity at the beginning of the `kick!` as
<img width="491" alt="grafik" src="https://github.com/user-attachments/assets/4236ff5a-4319-4fc5-80cc-4e105b8678fa">
In the first screenshot, you can see that dt/2 is used instead in the first stage, which is the "stage-local" step size used for that stage.
So we have two options:
1. Just use dt everywhere. It doesn't really matter that it is bigger than the stage-local step size, as the background pressure is arbitrary anyway.
2. Use `dt = t_stage - t_laststep`. The main difference to option 1 is that this dt is getting bigger for later stages, which is the same as in the first screenshot above, so this is closer (identical?) to what Ramachandran is doing. We could either fiddle with a stage limiter (which has the disadvantage that now all methods support it), or we could just store the time after the last full time step with a callback as `t_laststep`.

So then the transport velocity is not integrated, but locally computed at the beginning of `kick!` with the equation in the second screenshot and `dt = t - t_laststep`. This will also work for Euler.

Reply:
As discussed in zoom, the challenge is that $f^{n+1/2}$ depends on $u^{n+1/2}$ *and* $\tilde{u}^{n+1/2}$.
It is not as easy as it seems (or impossible?) with our time integrator approach. So far, the current implementation of this PR is the best solution or rather workaround. 
However, @efaulhaber will implement a more flexible time integrator approach, right? Does an issue already exist or will you create one?

Reply:
So we wait with this PR till this is finished?

Reply:
No, I would say that the current implementation is sufficient (according to the validation results). That is, we also mark this feature as experimental implementation. 

However, it would be interesting to compare the results with appropriated time integration schemes.